src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,167 @@
+package duke.parser;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+
+import duke.commands.Deadline;
+import duke.commands.Event;
+import duke.commands.Task;
+import duke.commands.Todo;
+import duke.data.TaskList;
+import duke.exceptions.DukeException;
+import duke.ui.Ui;
+
+/**
+ * Encapsulates a Parser object that reads in and executes commands based on user's input.
+ *
+ * @author Owen Tan
+ * @version duke.Duke Level-9
+ */
+public class Parser {
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Commands from user's input
+     */
+    public enum Keyword {
+        TODO, LIST, DEADLINE, EVENT, DONE, DELETE, FIND
+    }
+
+    /**
+     * Public constructor for Parser.
+     *
+     * @param tasks A list of tasks.
+     * @param ui Ui for duke.Duke.
+     */
+    public Parser(TaskList tasks, Ui ui) {
+        this.tasks = tasks;
+        this.ui = ui;
+    }
+
+    /**
+     * Reads in and executes user's input based on string given.
+     *
+     * @param cmd String from user's input.
+     * @throws DukeException
+     */
+    public String parse(String cmd) throws DukeException {
```

Comment:
This parse() method is a bit too long and there might be some violation of SLAP rule. Maybe we can consider extracting some String calculation, task initialization and task execution to form new methods for better abstraction.
Other than that I think you have done a really good job. The code is clean and neat.